library/src/commonMain/kotlin/com/ioki/passenger/api/models/AnyValue.kt

```diff
@@ -50,13 +60,25 @@ internal object AnyValueSerializer : KSerializer<AnyValue> {
 
         // Here we can add logic to determine the type
         return when {
-            jsonElement is JsonPrimitive && jsonElement.isString -> AnyValue.StringValue(jsonElement.content)
+            jsonElement is JsonPrimitive && jsonElement.isString -> runCatching {
+                AnyValue.InstantValue(input.json.decodeFromJsonElement(Instant.serializer(), jsonElement))
+            }.getOrElse {
+                AnyValue.StringValue(jsonElement.content)
+            }
+
             jsonElement is JsonPrimitive && jsonElement.intOrNull != null -> AnyValue.IntValue(jsonElement.int)
             jsonElement is JsonPrimitive && jsonElement.booleanOrNull != null ->
-                AnyValue.BooleanValue(
-                    jsonElement.boolean,
+                AnyValue.BooleanValue(jsonElement.boolean)
+
+            jsonElement is JsonPrimitive && jsonElement.doubleOrNull != null ->
+                AnyValue.DoubleValue(jsonElement.double)
+
+            jsonElement is JsonObject -> run {
```

Comment:
Run is not `runCatching?!` 🤔 
Just return the `AnyValue.ApiPointValue(...)`. It will throw if it fails.